File: go/tls_sig_api/tool/verify_token.go
package main

import (
	"flag"
	"log"

	"github.com/xverse-lab/tls-sig-api/go/tls_sig_api"
)

func main() {
	var token, sk string
	flag.StringVar(&token, "token", "", "token to verify")
	flag.StringVar(&sk, "sk", "", "secret key")
	flag.Parse()

	if token == "" || sk == "" {
		log.Fatalln("please input token and sk, you can input --help to get usage")
	}

	err := tls_sig_api.VerifyToken(token, sk)
	if err != nil {
		log.Fatalf("invalid token, %v", err)
	}
	log.Println("verify success")
}

File: cpp/tool/verify_token.cpp
#include "tls_token.h"
#include <iostream>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    if (argc != 3) {
        std::cout << "usage:\n" << argv[0] << "  token  sk\n";
        return -1;
    }

    string token = argv[1];
    string sk = argv[2];

    int ret = VerifyToken(token, sk);
    if (ret != 0) {
        std::cout << "token verify failed, " << ret << std::endl;
    } else {
        std::cout << "token verify success" << std::endl;
    }
    return 0;
}
